### PR TITLE
Fixes #1268 Add input parameter to wbemcli to allow mocking a server

### LINKED ISSUE #1268: Add mock capability to wbemcli
A new option on wbemcli that allows it to execute the mock wbemconnection with a defined input file.

### PR DESCRIPTION
**Pull request by KSchopmeyer**
This adds a parameter --mock_server to the command line parameters of
wbemcli to allow wbemcli to be used with the pywbem_mock environment.

This also adds one new constructor parameter to FakedWBEMConnection,
timeout that the WBEMConnection can use to return the timeout exception
if this value is set and is of a lower value than the response_delay
time which would indicate that the server delayed longer than the client
timeout setting

It also adds a test of the wbemcli new parameter to test_wbemcli.py and
a simple mof file to be used in that test.

Adds initial tests. Adds tests for the new parameters which means two new files in testsuite. This DOES NOT convert test_wbemcli.py to pytest.

Adds entry under enhancements to changes.rst

### CODE REVIEW

File: pywbem_mock/_wbemconnection_mock.py [hunk 1]
```diff
@@ -1421,7 +1425,7 @@ def _filter_properties(obj, property_list):
         Remove properties from an instance or class that aren't in the
         plist parameter
 
-        obj(:class:`~pywbem.CIMClassName` or :class:`~pywbem.CIMClassName):
+        obj(:class:`~pywbem.CIMClass` or :class:`~pywbem.CIMClass):
             The class or instance from which properties are to be filtered
```
Comment (andy-maier) [inline pywbem_mock/_wbemconnection_mock.py]: I assume this should have been CIMClass or CIMInstance?

Comment (KSchopmeyer) [reply pywbem_mock/_wbemconnection_mock.py]: Yes. FIXED.

**Comment by KSchopmeyer** [comment]
The bigger question might be a better way to incorporate python scripts than exec. That one does bother me and I am using it both here and in pywemcli.  It works but there must be a better way to get to plugins, etc.

**Comment by andy-maier** [comment]
Merging this PR because the question of executing scripts is not part of its changes. If you want to follow up on that, open an issue and try to phrase the issues with using exec().

**Pull request merged**